File: src/main/java/com/example/ajaxonboardingproject/service/mapper/OrderMapper.java
package com.example.ajaxonboardingproject.service.mapper;

import com.example.ajaxonboardingproject.dto.response.OrderResponseDto;
import com.example.ajaxonboardingproject.model.Order;
import com.example.ajaxonboardingproject.model.Ticket;
import org.springframework.stereotype.Component;

@Component
public class OrderMapper implements ResponseDtoMapper<OrderResponseDto, Order> {
    @Override
    public OrderResponseDto mapToDto(Order order) {
        return new OrderResponseDto(
                order.getId(),
                order.getTickets()
                        .stream()
                        .map(Ticket::getId)
                        .toList(),
                order.getUser().getId(),
                order.getOrderTime());
    }
}

File: src/main/java/com/example/ajaxonboardingproject/service/mapper/OrderMapper.kt
package com.example.ajaxonboardingproject.service.mapper

import com.example.ajaxonboardingproject.dto.response.OrderResponseDto
import com.example.ajaxonboardingproject.model.Order
import org.springframework.stereotype.Component

@Component
class OrderMapper : ResponseDtoMapper<OrderResponseDto, Order> {
    override fun mapToDto(model: Order): OrderResponseDto {
        return OrderResponseDto(
            id = model.id.toLong(),
            ticketIds = model.tickets
                .map { it.id.toLong() }
                .toMutableList(),
            userId = model.user.id.toLong(),
            orderTime = model.orderTime
        )
    }
}
